vowpalwabbit/feature_group.h

```diff
@@ -194,17 +194,23 @@ class features_value_index_audit_iterator : public features_value_index_iterator
   features_value_index_audit_iterator& operator++()
   {
     features_value_index_iterator::operator++();
-    _begin_audit++;
+    if (_begin_audit != nullptr)
```

Comment:
Can't we make sure we never construct an iterator with a null _begin_audit?

Reply:
The problem is that when you are using a cache file you don't really want to deal with audit strings and definitely care about speed. We could potentially template that those functions to allow that, but that is a larger chunk of work. In that case we would most likely just be inserting the same pairs of "","".